File: baekJoon/11689.py
import math


def isPrime(n):
    if n==2 or n==3:
        return True
    elif n%2 ==0 or n==1:
        return False
    for i in range(3, int(math.sqrt(n))+1, 2):
        if n%i == 0:
            return False
    return True

def nextPrime(n):
    n +=1
    while not isPrime(n):
        n+=1
    return n

def factorization(x):
    d = 2
    ans = {}
    while d <= x:
        if x % d == 0:
            if ans.get(d) == None:
                ans.update({d:1})
            else:
                ans[d]+=1
            x = x // d
            if isPrime(x):
                if ans.get(x) == None:
                    ans.update({x:1})
                else:
                    ans[x] +=1
                break
        else:
            d += 1
    return ans

ans = 1
a = factorization(int(input()))
print(a)
for k in a.keys():
    ans = ans * k**(a[k]-1) * (k-1)
print(ans)

File: baekJoon/11689.cpp
#include <iostream>
#include <math.h>
#include <map>
#include <stdlib.h>

using namespace std;
typedef unsigned long long U64;

bool isPrime(U64 n){
    if(n==2 || n==3)
        return true;
    else if(n%2 ==0 || n==1)
        return false;
    
    for(U64 i=3; i<U64(sqrtl(n)+1) ; i+=2){
        if (n%i == 0) return false;
    }
    return true;
}

int main(){
    map<U64, int> factor;

    U64 d = 2;
    U64 x ,ans = 1;
    cin >> x;
    if(isPrime(x)){
        cout << x-1 <<endl;
        return 0;
    }
    while(d<= x){
        if(x%d == 0){
            if(factor.find(d) == factor.end()){
                factor[d] = 1;
            }
            else{
                factor[d]++;
            }
            x = x/d;
            if (isPrime(x)){
                factor[x]++;
                break;
            }
        }
        else{
            d++;
        }
    }

    for(auto i=factor.begin(); i!=factor.end(); i++){
        // cout << i->first << " " << i->second << endl;

        ans = ans * U64(powl(i->first, i->second-1)) * (i->first-1);
    }

    cout << ans <<endl;
    return 0;
}